packages/atomic/src/components/result-template-components/result-template-decorators.tsx

```diff
@@ -58,3 +58,30 @@ export function ResultContext() {
     };
   };
 }
+
+type ResultContextEventHandler = (result: Result) => void;
+export type ResultContextEvent = CustomEvent<ResultContextEventHandler>;
+const resultContextEventName = 'atomic/resolveResult';
+
+/**
+ * Retrieves `Result` on a rendered `atomic-result`.
+ *
+ * This method is useful to build custom result template elements, see [Create a Result List](https://docs.coveo.com/en/atomic/latest/usage/create-a-result-list/) for more information.
+ *
+ * @param element Element on which to dispatch the event, which must be the child of a rendered "atomic-result".
+ * @returns A promise that resolves on initialization of the parent "atomic-result" element, and rejects when it's not the case.
+ */
+export const resultContext = (element: Element) =>
+  new Promise<Result>((resolve, reject) => {
+    setTimeout(() => {
```

Comment:
Something strange happening here...

Without the timeout, the result cannot be resolve, it seems that for an instant, the element exists somewhere not in the DOM, it must be linked to the way we render result templates

Reply:
Perhaps `initialize` is being called before [componentDidRender](https://stenciljs.com/docs/component-lifecycle#component-lifecycle-methods) in the case of result templates?

Reply:
It's called on connectedCallback in my example, which should be "Invoked each time the custom element is appended into a document-connected element."

To be more precise, the event would not resolve (wouldn't be caught as if the parent atomic-result did not exist). document.body.contain would be false too and element.closest('atomic-result') would return nothing as well

Reply:
```suggestion
    // TODO: https://coveord.atlassian.net/browse/KIT-127 make it work with connectedCallback
    setTimeout(() => {
```